### PR TITLE
Add AC Audio searcher for tutorial

### PR DESCRIPTION
**Pull request by huffman**
This is the same audio searcher we've been using in the tutorial for a long time, with a small tweak: the audio searcher will no longer spam the logs with messages about userData not being JSON, it will instead only log it the first time it finds such an entity.

### CODE REVIEW

File: tutorial/ACAudioSearchAndInject_tutorial.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+if (isACScript) {
+    EntityViewer.setCenterRadius(QUERY_RADIUS);
+}
+
+// ENTITY DATA CACHE
+//
+var entityCache = {}; // A dictionary of unexpired EntityData objects.
+var entityInvalidUserDataCache = {}; // A cache containing the entity IDs that have
+                                     // previously been identified as containing non-JSON userData.
+                                     // We use a dictionary here so id lookups are constant time.
```
Comment (huffman) [inline tutorial/ACAudioSearchAndInject_tutorial.js]: I added this variable

File: tutorial/ACAudioSearchAndInject_tutorial.js [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                var userData = properties.userData && JSON.parse(properties.userData);
+                soundData = userData && (userData[SOUND_DATA_KEY] || userData[old_sound_data_key]); // Don't store soundData yet. Let state changes compare.
+                that.lastUserDataUpdate = now;                    // But do update these ...
+                that.url = soundData && soundData.url;
+                that.playAfter = that.url && now;
+            } catch (err) {
+                if (!(entityIdentifier in entityInvalidUserDataCache)) {
+                    print(err, properties.userData);
+                    entityInvalidUserDataCache[entityIdentifier] = true;
+                }
```
Comment (huffman) [inline tutorial/ACAudioSearchAndInject_tutorial.js]: And this `if` block

**Comment by huffman** [comment]
I tested this

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e1f03db430ab5c97affffc45d04210d46a899ef available at: https://gist.github.com/04a271c5eb2005042bdb7147c1193019<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9103-8e1f03d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9103-8e1f03d.dmg

**Pull request merged**